Smart-Contract-Verifier-Server/src/main/kotlin/com/savannasolutions/SmartContractVerifierServer/messenger/services/MessengerService.kt

```diff
@@ -96,34 +96,55 @@ class MessengerService constructor(val messagesRepository: MessagesRepository,
+            status =  ResponseStatus.SUCCESSFUL)
     }
 
-    fun getMessageDetail(getMessageDetailRequest: GetMessageDetailRequest): GetMessageDetailResponse{
-        if(!messagesRepository.existsById(getMessageDetailRequest.MessageID))
-            return GetMessageDetailResponse(status = ResponseStatus.FAILED)
+    fun getMessageDetail(userID: String, messageID: UUID): ApiResponse<GetMessageDetailResponse>{
+        if(!messagesRepository.existsById(messageID))
+            return ApiResponse(status = ResponseStatus.FAILED,
+                message = commonResponseErrorMessages.messageDoesNotExist)
+
+        val message = messagesRepository.getById(messageID)
+        var found = false
+        if(message.sender.publicWalletID == userID)
+            found = true
+
+        val messageStatus = messageStatusRepository.getAllByMessage(message)
+        if(messageStatus != null) {
+            for (msgStatus in messageStatus)
+            {
+                if(msgStatus.recipient.publicWalletID == userID)
+                    found = true
+            }
+        }
+
+        if(!found)
+            return ApiResponse(status = ResponseStatus.FAILED,
+                message = commonResponseErrorMessages.userNotPartOfAgreement)
 
-        val message = messagesRepository.getById(getMessageDetailRequest.MessageID)
         val messageList = ArrayList<Messages>()
         messageList.add(message)
         val messageResponseList = generateMessageResponseList(messageList)
 
-        return GetMessageDetailResponse(messageResponseList[0], status = ResponseStatus.SUCCESSFUL)
+        return ApiResponse(responseObject = GetMessageDetailResponse(messageResponseList[0]),
+            status = ResponseStatus.SUCCESSFUL)
     }
 
-    fun sendMessage(sendMessageRequest: SendMessageRequest): SendMessageResponse{
-        //TODO Implement setting jury as recipients as well
-        if(!userRepository.existsById(sendMessageRequest.SendingUser))
-            return SendMessageResponse(status = ResponseStatus.FAILED)
+    fun sendMessage(userID:String, agreementID:UUID, sendMessageRequest: SendMessageRequest): ApiResponse<SendMessageResponse>{
+        if(!userRepository.existsById(userID))
+            return ApiResponse(status = ResponseStatus.FAILED,
+                        message = commonResponseErrorMessages.userDoesNotExist)
 
-        if(!agreementsRepository.existsById(sendMessageRequest.AgreementID))
-            return SendMessageResponse(status = ResponseStatus.FAILED)
+        if(!agreementsRepository.existsById(agreementID))
+            return ApiResponse(status = ResponseStatus.FAILED,
+                        message = commonResponseErrorMessages.agreementDoesNotExist)
 
         if(sendMessageRequest.Message.isEmpty())
-            return SendMessageResponse(status = ResponseStatus.FAILED)
+            return ApiResponse(status = ResponseStatus.FAILED, message = "Message is empty")
```

Comment:
Didn't you create common error messages?

Reply:
So this is a new feature. In stead of just saying failed or successful we can now actually see what failed. I do have a class with common messages, but thus is an uncommon error message